docs/block-processing/overall.md

```diff
@@ -0,0 +1,107 @@
+# Block processing
+
+## Task definition
+
+Block processing can be described by presenting an algorithm to solve the following problem:
+
+> Given a sequence of blocks `B₀, B₁, B₂, …` 
+(where `B₀` is the first genesis block) check whether block payload from these blocks are valid.
+
+We describe block processing as stateful algorithm:
+* Initial state `S₀` is derived from blockchain genesis data (see [mainnet genesis JSON](https://raw.githubusercontent.com/input-output-hk/cardano-sl/e7cfb1724024e0db2f25ddd2eb8f8f17c0bc497f/node/mainnet-genesis.json))
+* `S₁, S₂, …` are maintained as sequential application of blocks  `B₀, B₁, B₂, … ` to state `S₀`
+* State transition function. 
+We maintain some state called GState which corresponds to the application of a sequence of blocks
```

Comment:
I suggest that you move this sentence above and make it a separate bullet point (the first one). So that it appears before 
> Initial state `S₀` is derived from blockchain genesis data…

Also you forgot a dot.

Reply:
I meant moving it before 
> Initial state `S₀` is derived from blockchain genesis data (see [mainnet genesis JSON](https://raw.githubusercontent.com/input-output-hk/cardano-sl/e7cfb1724024e0db2f25ddd2eb8f8f17c0bc497f/node/mainnet-genesis.json))

Then it will be clear that `S₀` is actually GState.